asm_utils/src/data_storage.rs

```diff
@@ -0,0 +1,80 @@
+//! Utilites for generating initialization code that stores data objects in memory.
+
+use std::collections::BTreeMap;
+
+use crate::{data_parser::DataValue, utils::next_multiple_of_four};
+
+pub enum SingleDataValue<'a> {
+    Value(u32),
+    LabelReference(&'a String),
+    Offset(&'a String, &'a String),
+}
+
+pub fn store_data_objects<'a>(
+    objects: impl IntoIterator<Item = &'a (String, Vec<DataValue>)> + Copy,
+    mut memory_start: u32,
+    code_gen: &mut dyn FnMut(u32, SingleDataValue) -> Vec<String>,
```

Comment:
Do we need dynamic dispatch here? Could it use a method in the `Architecture` trait or is it unrelated?

Reply:
That's a good point! If it is not capturing anything, it can just be a mehtod.